sbomify/apps/core/apis.py

```diff
@@ -1335,6 +1337,12 @@ def patch_project(request: HttpRequest, project_id: str, payload: ProjectPatchSc
                             )
                         }
 
+                # Enforce scope exclusivity: If assigning to a project, components cannot be global
+                # We automatically demote them from global scope
+                global_component_ids = [c.id for c in components if c.is_global]
```

Comment:
Please use better variable names. We don't really gain anything from using `c` over `component` other than lost clarity.